### ISSUE TITLE
Text set instantly when defined as right-button tool in mouse.

### ISSUE THREAD

**Issue by hiperbolt** [xournalpp/xournalpp#3758]
**Affects versions :**
 - OS: Fedora Workstation 35
 - Gnome-Wayland
 - libgtk version 3.24.31
 - Version of Xournal++: 1.1.0 
 - Installation method: Official Release

**Describe the bug**
Not sure if this is a bug or a feature request;
If "text" is set as as the right-button tool in mouse, it *should* wait until you finish typing it to be set. As it stands, as soon as you let go of the mouse, the text is instantly set and you can't keep typing.

**To Reproduce**
1. Set "text" as the right-button tool for the mouse in Edit->Preferences
2. Click right-button on mouse
3. Try to type.

**Expected behavior**
This is debatable, but from a usability stand-point, I think the expected behaviour would be that the text would be set and the tool would change back to the previously used one when the user hits enter, not when you let go of the right click button.

**Additional context**
I'll try to add a PR to change this, however, I'd like to see some opinions on it.

**Comment by rolandlo** [comment]
Text boxes can contain several lines (and hitting enter starts a new line), so I am not sure your solution is that good. I agree, however, that the current behaviour (although logical) makes it pretty hard to use the text tool assigned to a stylus button.

**Comment by bhennion** [comment]
A solution would be to use a modifier when the user wants to add a new line. 
See for instance the behaviour of `libreoffice calc`: `ENTER` passes to the next row, `CTRL+ENTER` adds a new line in the current cell. Similar behaviours are often found in IM apps: `ENTER` sends the message, `CTRL+ENTER` or `SHIFT+ENTER` adds a new line to the message.

So for us, it seems reasonible to me that:

- `ENTER` would stop editing the current text box.
- `CTRL+ENTER` would add a new line in the current text box.

**Comment by hiperbolt** [comment]
> A solution would be to use a modifier when the user wants to add a new line. See for instance the behaviour of `libreoffice calc`: `ENTER` passes to the next row, `CTRL+ENTER` adds a new line in the current cell. Similar behaviours are often found in IM apps: `ENTER` sends the message, `CTRL+ENTER` or `SHIFT+ENTER` adds a new line to the message.
> 
> So for us, it seems reasonible to me that:
> 
>     * `ENTER` would stop editing the current text box.
> 
>     * `CTRL+ENTER` would add a new line in the current text box.

Yes, I agree with your perspective. Although I'd suggest `SHIFT+ENTER` since that currently already works to add a new line to the current textbox, while  `CTRL+ENTER` does not.

I have no experience with the codebase but I'll try to submit a PR implementing this feature.

**Comment by moy** [comment]
Actually, this is the behavior of the text tool when used with the left mouse button (click to start entering text, then type text using <kbd>shift</kbd>+<kbd>Enter</kbd> for newlines, and validate using <kbd>Enter</kbd>). It's weird that the behavior is different when using another mouse button, or the stylus button.
